libp2p/protocols/pubsub/timedcache.nim

```diff
@@ -55,14 +74,14 @@ func put*[K](t: var TimedCache[K], k: K, now = Moment.now()): bool =
   # refreshed.
   t.expire(now)
 
-  var previous = t.del(k) # Refresh existing item
-
-  var addedAt = now
-  previous.withValue(previous):
-    addedAt = previous.addedAt
+  let
+    previous = t.del(k) # Refresh existing item
+    addedAt = if previous.isSome():
```

Comment:
We had a long PR in the past to remove this pattern from the codebase and decrease the risk of raising defects. You can use https://github.com/vacp2p/nim-libp2p/blob/2725be64bacb46997c0959560d833ff946c8cb3b/libp2p/utility.nim#L124.

Reply:
`valueOr` is not applicable in this case because we're accessing a field of `previous[]`, not `previous` itself

Reply:
True, but withValue can be used in this case.

Reply:
doesn't work in generic code, due to similar problems as https://github.com/arnetheduck/nim-results/issues/34

Reply:
this seems to work fine:

```
    addedAt = block:
      previous.withValue(p):
        p[].addedAt
      else:
        now
```